upup/pkg/fi/cloudup/awstasks/network_load_balancer_attachment.go

```diff
@@ -0,0 +1,160 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package awstasks
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+	"github.com/aws/aws-sdk-go/service/elbv2"
+	"k8s.io/klog"
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/cloudup/awsup"
+)
+
+//go:generate fitask -type=NetworkLoadBalancerAttachment
+type NetworkLoadBalancerAttachment struct {
+	Name      *string
+	Lifecycle *fi.Lifecycle
+
+	LoadBalancer *NetworkLoadBalancer
+
+	// LoadBalancerAttachments now support ASGs or direct instances
+	AutoscalingGroup *AutoscalingGroup
+	Subnet           *Subnet
+
+	// Here be dragons..
+	// This will *NOT* unmarshal.. for some reason this pointer is initiated as nil
+	// instead of a pointer to Instance with nil members..
+	Instance *Instance
```

Comment:
I can't think of any reason why we'd want to be attaching individual instances to an NLB so lets remove this field and any references to it. It would be nice to figure out why it existed in the ELB tasks too, but we can do that later.